### PR TITLE
Setup Redux

### PR DESCRIPTION
**Pull request by ka1130**
This PR sets up Redux and moves fetching responsibility to relevant reducers.

### CODE REVIEW

**Comment by ka1130** [comment]
Thanks for the tip, @yakato , I will read about it and add it then.

File: src/components/App/index.js [hunk 1]
```diff
@@ -47,11 +47,13 @@ class App extends Component {
       .catch(error => this.setState({ loading: false, error }));
   }
 
-  simpleAction = (event) => {
-    this.props.simpleAction();
+  fetchEpisodes = (event) => {
```
Comment (yakato) [inline src/components/App/index.js]: you probably don't need `event` here

File: src/redux/reducers/index.js [hunk 2]
```diff
@@ -1,8 +1,8 @@
 import { combineReducers } from 'redux';
-import simpleReducer from 'redux/reducers/simpleReducer';
+import fetchEpisodes from 'redux/reducers/fetchEpisodes';
```
Comment (yakato) [inline src/redux/reducers/index.js]: names of reducers should be more about data which they refer to, so in `episodes.js` reducers you should take care of actions like `FETCH_EPISODES_REQUEST`, ``FETCH_EPISODES_FAILURE`, `FETCH_EPISODES_SUCCESS` and maybe more in the future

File: src/redux/actions/fetchEpisodes.js [hunk 3]
```diff
@@ -1,12 +1,7 @@
-// export function fetchEpisodes() {
-//   return (dispatch) => {
-//     dispatch({ type: '' })
-//   }
-// }
-
 export const fetchEpisodes = () => dispatch => {
```
Comment (yakato) [inline src/redux/actions/fetchEpisodes.js]: Same approach as for reducers should be for actions. File name I can suggest is 'episodesActions' or 'episodes'. [Here](https://daveceddia.com/where-fetch-data-redux/) is an example of such a file and how to fetch data in action.

Also [actions doc](https://redux.js.org/advanced/asyncactions)

File: src/redux/reducers/fetchEpisodes.js [hunk 4]
```diff
@@ -7,4 +7,4 @@ export default (state = {}, action) => {
    default:
     return state
   }
- }
\ No newline at end of file
+ }
```
Comment (yakato) [inline src/redux/reducers/fetchEpisodes.js]: If I can suggest to plan store structure - name of file will be also name of key in the store, that's why 'episodes'. Please re-think `result` name of key. Your structure look like that now: 
```
store = {
   fetchEpisodes: {
      result: {payloadGoesHere}
   }
}
```

**Comment by ka1130** [comment]
Thanks, @yakato :) I've made some order in the files, please have a look and advice if I chose a convenient pattern. The fetching is now done via reducers. Hope I didn't mess up anything on the way but all seems to be up and running now.

File: src/redux/actions/episodesActions.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+      .then(res => res.json())
+      .then(json => {
+        dispatch(fetchEpisodesSuccess(json));
+        return json;
+      })
+      .catch(error => dispatch(fetchEpisodesFailure(error)));
+  };
+}
+
+function handleErrors(response) {
```
Comment (yakato) [inline src/redux/actions/episodesActions.js]: Where are you using this function?

File: src/redux/reducers/episodesReducer.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
+import {
+  FETCH_EPISODES_BEGIN,
+  FETCH_EPISODES_SUCCESS,
+  FETCH_EPISODES_FAILURE
+} from 'redux/actions/types';
+
+const initialState = {
+  episodes: [],
```
Comment (yakato) [inline src/redux/reducers/episodesReducer.js]: Re-think this name of key. Now your state looks like this:
screen shot 2018-10-05 at 07 55 03
And you can see there is episodes inside episodes. Maybe change into 'data' or 'all' ?

File: src/redux/reducers/index.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
+import { combineReducers } from 'redux';
+import fetchEpisodes from 'redux/reducers/episodesReducer';
```
Comment (yakato) [inline src/redux/reducers/index.js]: Re-think name of this reducer. There are two schools: one is create a couple of reducers in file 'episodesReducer' and then import all of them here. The other one is to name this reducer inside 'episoderReducer' or whatever and keep all cases inside, then there is no need to specify what you're importing and just say:
```
import episodes from 'redux/reducers/episodesReducer';
const rootReducer = combineReducers({
  episodes
});
```

Up to you which one you'll choose. In bigger apps I prefer second (when there is more than 2-3 reducers).

**Pull request merged**